src/buildApplication/checkPackageManager.ts

```diff
@@ -0,0 +1,50 @@
+import YAML from 'js-yaml';
+import { spawn } from 'child_process';
+import { readFile } from 'fs/promises';
+import { validateFile } from '../utils';
+import { cliError } from '../cli';
+
+export async function checkPackageManager() {
+	const userAgent = process.env.npm_config_user_agent;
+
+	const hasYarnLock = await validateFile('yarn.lock');
+	const hasPnpmLock = await validateFile('pnpm-lock.yaml');
+
+	if ((userAgent && userAgent.startsWith('pnpm')) || hasPnpmLock) return 'pnpm';
+
+	if ((userAgent && userAgent.startsWith('yarn')) || hasYarnLock) {
+		const yarn = process.platform === 'win32' ? 'yarn.cmd' : 'yarn';
```

Comment:
I think that we can have a utility function in the utils folder, based on my previous comment I would suggest something like this:
![Screenshot 2023-04-05 at 09 53 16](https://user-images.githubusercontent.com/61631103/230032038-e2dd2ea5-29f0-4724-996a-3a9ffb8d6ccd.png)

so you don't need to do the window checking every time, and this line would become
```diff
-		const yarn = process.platform === 'win32' ? 'yarn.cmd' : 'yarn';
+               const yarn = getSpawnCommand('yarn');
```